pq-crypto/EVP_kem.h

```diff
@@ -0,0 +1,179 @@
+* Name: pq_kem_params_free
+*
+* Arguments: pointer to pq_key_params.
+* pq_key_params allows access to the lenghts and key pointer (output param)
+ * of public key, private key, cipher text,
+* and shared secret.
+* Description: Frees space of
+* public key, private key, cipher text,
+* and shared secret.
+*
+* Return EVP_PQ_SUCCESS on success, and EVP_PQ_FAILURE if it fails.
+**************************************************/
+int pq_kem_params_free(pq_kem_params *key_params);
+
+/*************************************************
+* Name: EVP_kem_generate_keypair
+*
+* Description: Generates a public and private key
+*
+* Arguments: pq_kem_params
+* The following fields of pq_kem_params are used for generate keypair:
+* - unsigned char *public_key: pointer to output public key
+* (already allocated array of bytes)
+* - unsigned char *private_key: pointer to output secret key
+* (already allocated array of bytes)
+*
+* Returns EVP_PQ_SUCCESS on successfully generating key pair,
+* return EVP_PQ_FAILURE otherwise and on error
+**************************************************/
+int EVP_kem_generate_keypair(pq_kem_params *kem_params);
+
+
+/*************************************************
+* Name: EVP_kem_enc
+*
+* Description: Uses public key to create cipher text and secrect key.
+*
+* Arguments: pq_kem_params
+* The following fields of pq_kem_params are used for encapsualte:
+* - unsigned char *cipher_text: pointer to output cipher text
+* (already allocated array of bytes)
+* - unsigned char *shared_secret: pointer to output shared secrect
+* (already allocated array of bytes)
+* - const unsigned char *public_key: pointer to input constant public key
+*
+* Returns EVP_PQ_SUCCESS on successful encapsulation,
+* return EVP_PQ_FAILURE otherwise and on error
+**************************************************/
+int EVP_kem_encapsulate(pq_kem_params *kem_params);
+
+
+/*************************************************
+* Name: EVP_kem_dec
+*
+* Description: Generates shared secret for public key and cipher text
+*
+* Arguments: pq_kem_params
+* The following fields of pq_kem_params are used for decapsulate:
+* - unsigned char *shared_secret: pointer to output shared secret
+* (already allocated array of bytes)
+* - unsigned char *cipher_text: pointer to input cipher text
+* (already allocated array of bytes)
+* - const unsigned char *private_key: pointer to input constant private key
+*
+* Returns EVP_PQ_SUCCESS on successful decapsulation,
+* return EVP_PQ_FAILURE otherwise and on error
+**************************************************/
+int EVP_kem_decapsulate(pq_kem_params *kem_params);
+
+
+/* SIKE's key generation
+ * Outputs: private key (SIKE_P434_R3_SECRET_KEY_BYTES = SIKE_P434_R3_MSG_BYTES + SIKE_P434_R3_SECRETKEY_B_BYTES + SIKE_P434_R3_PUBLIC_KEY_BYTES bytes)
+ *          public key (SIKE_P434_R3_PUBLIC_KEY_BYTES bytes) */
+int evp_sike_p434_r3_crypto_kem_keypair(OUT unsigned char *public_key, OUT unsigned char *private_key);
```

Comment:
I think the "evp" naming convention only applies to functions that make up the public API. We either want to expose the above keypair/encapsulate/decapsulate functions, or these functions, but probably not both. Which is to say: I don't think these functions need to follow the "evp" convention, nor do I think they need to be in the "public" header.

Reply:
I think I still maintain this opinion :P